src/Container.php

```diff
@@ -665,6 +665,129 @@ public function share($key, $value, $protected = false)
         return $this->set($key, $value, true, $protected);
     }
 
+    /**
+     * Create a lazy proxy resource for given class, and register it in the Container.
+     *
+     * @param   string         $class      Full class name of the resource.
+     * @param   callable       $factory    Optional callback to create the class instance. The callback must return instance of the given class.
+     *                                     When provided then Lazy Proxy object will be created, otherwise Lazy Ghost will be created.
+     * @param   boolean        $shared     True to create and store a shared instance.
+     * @param   boolean        $protected  True to protect this item from being overwritten. Useful for services.
+     *
+     * @return  $this
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    final public function lazy1(string $class, callable $factory, bool $shared = false, bool $protected = false): static
+    {
+        $featureSupported = PHP_VERSION_ID >= 80400;
+
+        // Create a Lazy Proxy factory
+        $lazyFactory = $featureSupported ? function () use ($class, $factory) {
+            return (new \ReflectionClass($class))->newLazyProxy(function () use ($factory) {
+                return $factory($this);
+            });
+        } : $factory;
+
+        return $this->set($class, $lazyFactory, $shared, $protected);
```

Comment:
I would use early return for this.

```suggestion
        if (PHP_VERSION_ID < 80400) {
            return $this->set($class, $Factory, $shared, $protected);
        }

        // Create a Lazy Proxy factory
        $lazyFactory = function () use ($class, $factory) {
            return (new \ReflectionClass($class))->newLazyProxy(function () use ($factory) {
                return $factory($this);
            });
        };

        return $this->set($class, $lazyFactory, $shared, $protected);
```